SdlLogs/SdlLogs.py

```diff
@@ -223,25 +225,61 @@ def run(self, edit):
                 there is value in 'source_path' attribute in settings file
                 there isn't value in 'source_path' attribute in settings file
         """
-        FULL_PATH_FILE = "(/.*)(/sdl_core/src/.{0,}\\.(cc|h|cpp|hpp)):(\\d{1,})"
-        CORRECTED_PATH = "(/[^/ ]*)+/?"
-        primaryPath = re.search(FULL_PATH_FILE, self.view.substr(
-            self.view.line(self.view.sel()[0])))
-        isItPath = re.search(CORRECTED_PATH, self.view.substr(
+
+        def open_file(self, filePath):
+            """
+                The function hides the function call to jump to file and to the desired line.
+            """
+            self.view.window().open_file(filePath, sublime.ENCODED_POSITION)
+
+        SDL_LOG_SOURCE = "/sdl_core/src/"
+
+        FILE_NAME_WITHOUT_PATH_NAME = "(.+)(/|(] +))(.{0,}\\.(cc|h|cpp|hpp)):(\\d{1,})"
+        FULL_PATH_FILE = "(/.*)(/sdl_core/src/.{0,}\\.(c|cc|h|cpp|hpp)):(\\d{1,})"
+
+        # Check exist file in directory
+        # file in directory have
+        primaryPath_exist = re.search(
+            FULL_PATH_FILE, self.view.substr(self.view.line(self.view.sel()[0])))
+        # file in directory haven't
+        path_not_exist = re.search(FILE_NAME_WITHOUT_PATH_NAME, self.view.substr(
             self.view.line(self.view.sel()[0])))
-        if primaryPath:
+
+        if primaryPath_exist:
+            if primaryPath_exist:
+                file = primaryPath_exist.group(
+                    1) + primaryPath_exist.group(2)
+
+                if path.isfile(file):
+                    open_file(self, file + ":" + primaryPath_exist.group(4))
```

Comment:
@MaximRybak please extract from regexp groups to human readable variables

Reply:
Fixed in c82425a84f05a787910dbe0681d7d4b1f6a15a91